pkg/controller/v1beta1/inferenceservice/reconcilers/kubernetes/kube_reconciler.go

```diff
@@ -0,0 +1,387 @@
+										ServicePort: intstr.IntOrString{
+											Type:   intstr.Int,
+											IntVal: podSpec.Containers[0].Ports[0].ContainerPort,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	return ingress
+}
+
+func getClientInCluster() (*kubernetes.Clientset, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	return kubernetes.NewForConfig(config)
+}
+
+//Reconcile ...
+func (r *NativeReconciler) Reconcile() (*appsv1.Deployment, error) {
+	client, err := getClientInCluster()
+	if err != nil {
+		return nil, err
+	}
+
+	//reconcile Deployment
+	deployment := &appsv1.Deployment{}
+	checkResult, err := r.checkNativeDeploymentExist(client)
+	switch {
+	case checkResult == CheckResultCreate:
+		deployment, err = client.AppsV1().Deployments(r.Deployment.Namespace).Create(context.TODO(), r.Deployment, metav1.CreateOptions{})
+		if err != nil {
+			return nil, err
+		}
+	case checkResult == CheckResultExisted:
+		deployment = r.Deployment
+	case checkResult == CheckResultUpdate:
+		deployment, err = client.AppsV1().Deployments(r.Deployment.Namespace).Update(context.TODO(), r.Deployment, metav1.UpdateOptions{})
+		if err != nil {
+			return nil, err
+		}
+	case checkResult == CheckResultUnknown:
+		return nil, err
+	}
+
+	//reconcile Service
+	checkResult, err = r.checkNativeServiceExist(client)
+	switch {
+	case checkResult == CheckResultCreate:
+		_, err = client.CoreV1().Services(r.Service.Namespace).Create(context.TODO(), r.Service, metav1.CreateOptions{})
+		if err != nil {
+			return nil, err
+		}
+	case checkResult == CheckResultExisted:
+		log.Info("Reconcile", "Service", "Existed. Will do nothing.")
+	case checkResult == CheckResultUpdate:
+		_, err = client.CoreV1().Services(r.Service.Namespace).Update(context.TODO(), r.Service, metav1.UpdateOptions{})
+		if err != nil {
+			return nil, err
+		}
+	case checkResult == CheckResultUnknown:
+		return nil, err
+	}
+
+	//reconcile Ingress
+	checkResult, err = r.checkNativeIngressExist(client)
+	switch {
+	case checkResult == CheckResultCreate:
+		_, err = client.ExtensionsV1beta1().Ingresses(r.Ingress.Namespace).Create(context.TODO(), r.Ingress, metav1.CreateOptions{})
+		if err != nil {
+			return nil, err
+		}
+	case checkResult == CheckResultExisted:
+		_ = r.Ingress
+	case checkResult == CheckResultUpdate:
+		_, err = client.ExtensionsV1beta1().Ingresses(r.Ingress.Namespace).Update(context.TODO(), r.Ingress, metav1.UpdateOptions{})
+		if err != nil {
+			return nil, err
+		}
+	case checkResult == CheckResultUnknown:
+		return nil, err
+	}
+	return deployment, nil
+}
+
+//checkNativeDeploymentExist checks is the deployment is existed?
+func (r *NativeReconciler) checkNativeDeploymentExist(client *kubernetes.Clientset) (CheckResultType, error) {
+	//get deployment
+	existingDeployment, err := client.AppsV1().Deployments(r.Deployment.Namespace).Get(context.TODO(), r.Deployment.Name, metav1.GetOptions{})
```

Comment:
Can we use the controller runtime [client API](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/client.go)? want to be consistent with other places.